.github/workflows/release.yml

```diff
@@ -0,0 +1,152 @@
+# This workflow takes care of creating release archives for the
+# Flint distribution. It is run for all PR and branch pushes as usual,
+# but also on tags whose name starts with `vX.Y` with X, Y numbers
+# (the idea is to use v1.2.3 or v1.2.3-beta3)
+#
+# For builds triggered by a tag, the tag is turned into a GitHub release and
+# the produced archives are attached to that.
+name: "Wrap releases"
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    tags:
+      - v[1-9]+.[0-9]+.[0-9]    # allow v1.2.3
+      - v[1-9]+.[0-9]+.[0-9]-*  # allow v1.2.3-beta3 etc.
+    branches:
+      - trunk
+      - flint-*
+  schedule:
+    # Every day at 3:33 AM UTC
+    - cron: '33 3 * * *'
+
+concurrency:
+  # group by workflow and ref; the last slightly strange component ensures that for pull
+  # requests, we limit to 1 concurrent job, but for the trunk branch we don't
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.ref != 'refs/heads/trunk' || github.run_number }}
+  # Cancel intermediate builds, but only if it is a pull request build.
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+
+jobs:
+  make-archive:
+    runs-on: ubuntu-latest
+    outputs:
+      get-version: ${{ steps.get-version.outputs.version }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: "Setup"
+        run: |
+          sudo apt install -y libgmp-dev
+          sudo apt install -y libmpfr-dev
+          sudo apt install -y autoconf
+          sudo apt install -y libtool-bin
+          gcc --version
+          gcov --version
+          make --version
+          autoconf --version
+          libtool --version
+
+      # special treatment for tags: these are used for actual releases,
+      # so enforce that the version in configure.ac and in the tag match
+      - name: "Patch configure.ac"
+        if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+        run: |
+          version=${GITHUB_REF#refs/tags/v}
+          echo "version=$version"
+          sed -i -e "s/define(FLINT_VERSION,.*/define(FLINT_VERSION,[$version])/" configure.ac
```

Comment:
I don't know how github workflows work, but I'm a bit confused by this. Are you patching configure.ac a first time here, and again as part of `make dist`?